app/src/main/java/io/gnosis/safe/ui/safe/settings/safe/AdvancedSafeSettingsViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package io.gnosis.safe.ui.safe.settings.safe
+
+import io.gnosis.data.models.SafeInfo
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import pm.gnosis.model.Solidity
+import javax.inject.Inject
+
+class AdvancedSafeSettingsViewModel
+@Inject constructor(
+    private val safeRepository: SafeRepository,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<AdvancedSafeSettingsState>(appDispatchers) {
+
+    override fun initialState(): AdvancedSafeSettingsState = AdvancedSafeSettingsState()
+
+    fun load() {
+        safeLaunch {
```

Comment:
so initially there will be no progress indicator shown even though we are requesting recent safe info details from backend?

Reply:
The default value is `true` for `loading` in the `AdvancedSafeSettingsState` . Check out line 40 in this file.

Reply:
This is set to loading with the `override fun initialState()` method